src/s3ql/backends/b2/b2_backend.py

```diff
@@ -662,6 +665,16 @@ def _invalidate_upload_url(self, upload_url_info):
 
         log.debug('upload urls left: %d', len(self.available_upload_url_infos))
 
+    # TODO: schedule to disconnect connections as we use them, don't leave lingering connections until next request comes
+    def _expire_upload_url_info(self, upload_url_info):
+        '''Disconnects an active connection that has been unused for too long.'''
+        if (upload_url_info['connection'] is not None
+                and upload_url_info['lastUsed'] != 0
+                and time.monotonic() > upload_url_info['lastUsed'] + 10):
```

Comment:
10 seconds feels pretty random. Maybe use the tcp-timeout backend option instead? Or introduce a new option?